client/app/scripts/liveblog-advertising/controllers/advertising.js

```diff
@@ -96,18 +108,128 @@ upload, $templateCache, freetypeService, modal) {
         if ($scope.advert._id) {
             // we are editing existing ad
             api('advertisements').save($scope.advert, newAd).then(function(data) {
-                handleSaveSuccess();
+                handleAdvertSaveSuccess();
             }, function(data) {
-                handleSaveError();
+                handleAdvertSaveError();
             });
         } else {
             api('advertisements').save(newAd).then(function(data) {
-                handleSaveSuccess();
+                handleAdvertSaveSuccess();
             }, function(data) {
-                handleSaveError();
+                handleAdvertSaveError();
             });
         }
     }
 
     loadAdverts();
+
+
+    //COLLECTIONS
+    function loadCollections() {
+        $scope.collectionsLoading = true;
+        api('collections').query({where: {deleted: false}}).then(function(data) {
+            $scope.collections = data._items;
+            notify.info('Collections loaded');
+            $scope.collectionsLoading = false;
+        }, function(data) {
+            $scope.collectionsLoading = false;
+            notify.error(gettext('There was an error getting the adverts'));
+        })
+    }
+
+    function handleCollectionSaveSuccess() {
```

Comment:
no it shouldn't be a promise.
is how it says a handler that is called in the success promise.